src/beatree/leaf/free_list.rs

```diff
@@ -192,12 +181,14 @@ impl FreeList {
     ///   The old head will then contain only 4 elements, and the new head will contain one element, resulting in:
     ///
     /// [(Page number: 2, Free list: [1, 7, 10, 12]), (Page number: 9, Free list: [15])]
+    ///
+    /// It returns a Vec containing all the pages that need to be written into storage to reflect the current
+    /// status of the free list along with their page numbers
```

Comment:
Nit: I think this is not the ideal place. If you want to quickly look up what the function returns, you would look in the beginning of the rustdoc. At least that's the convention anyway: the important stuff comes first (with the first line reserved for the most important summary which is specially treated by dev software) and the details come later.